src/operator/image/image_random.cu

```diff
@@ -99,18 +111,134 @@ void ToTensorImplCUDA(mshadow::Stream<gpu> *s,
         W = input.size(2);
         C = input.size(3);
         blocks = N > 0 ? N : 1;
-        blocks = N;
     }
-    // One block per image.
-    // Number of threads = (32, 32) is optimal, because,
-    // computation is minimal and overhead of CUDA preparing
-    // all threads is minimal.
+
     ToTensorCudaKernel<gpu, DType>
-            <<<blocks, dim3(32, 32), 0, stream>>>(input, output,
+            <<<blocks, dim3(H, cuda::kMaxThreadsPerBlock / H), 0, stream>>>(input, output,
```

Comment:
please fix ToTensor similarly in a separate PR.

Reply:
Sure already work in progress. Should this PR wait till then or undo dim3(H, cuda::kMaxThreadsPerBlock / H) back to dim3(32, 32). Let me know next steps for this PR. Thanks!

Reply:
back to (32, 32), we can address it later.